src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Overview/ThorchainSaversOverview.tsx

```diff
@@ -99,23 +99,36 @@ export const ThorchainSaversOverview: React.FC<OverviewProps> = ({
   const highestBalanceAccountId = useAppSelector(state =>
     selectHighestBalanceAccountIdByStakingId(state, highestBalanceAccountIdFilter),
   )
-  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, ethChainId))
-  const opportunityDataFilter = useMemo(
-    () => ({
+  const defaultAccountId = useAppSelector(state => selectFirstAccountIdByChainId(state, chainId))
+  const maybeAccountId = useMemo(
+    () => accountId ?? highestBalanceAccountId ?? defaultAccountId,
+    [accountId, defaultAccountId, highestBalanceAccountId],
+  )
+
+  useEffect(() => {
+    if (!maybeAccountId) return
+    handleAccountIdChange(maybeAccountId)
+  }, [handleAccountIdChange, maybeAccountId])
```

Comment:
The meat of this PR, calls `onAccountIdChange()` same as if we were in a regular overview modal, not an empty one.
Generalizing this over all DeFi modal steps may be all we need to unrug deep linking to all steps